deploy/eck-stack/charts/eck-fleet-server/examples/fleet-server.yaml

```diff
@@ -0,0 +1,19 @@
+version: 8.17.0-SNAPSHOT
+deployment:
+  replicas: 1
+  podTemplate:
+    spec:
+      serviceAccountName: fleet-server
```

Comment:
There is a disconnect here between the name of this service account and the service account that is being generated. 

We are also setting the generated service account name as the `.spec.serviceAccountName` which is distinct from the service account in the pod spec. The latter is required for agent to work correctly the former is only relevant for the cross-namespace RBAC feature we have built into ECK ("Am I allowed to associated with an Elasticsearch in namespace x") . I am not sure if it is a problem to combine the two into one service account. Curious to get @barkbay 's perspective.

Reply:
At the very least the service account names in this example have to be the same for the example to work.

Reply:
I've updated this to make them consistent. What I tested was this: https://github.com/naemono/cloud-on-k8s/blob/helm-chart-image-fix-fleet/deploy/eck-stack/examples/agent/fleet-agents.yaml, which worked without issues, which only includes the sa name in the `podTemplate`.

Reply:
I'm not sure that's a problem either, that being said we made the choice to let the user specify a SA which can be different from the one used by the Pods (https://www.elastic.co/guide/en/cloud-on-k8s/master/k8s-restrict-cross-namespace-associations.html), and I'm not sure it is a reflected in the chart. [This comment](https://github.com/elastic/cloud-on-k8s/blob/f07d51dbe3a649923485511c2f18a421af2bd454/deploy/eck-stack/charts/eck-fleet-server/values.yaml#L104C1-L106) suggests that `serviceAccount.name` is used by the Pods while it is actually used by the [cross-namespace restriction mechanism](https://github.com/elastic/cloud-on-k8s/blob/f07d51dbe3a649923485511c2f18a421af2bd454/deploy/eck-stack/charts/eck-fleet-server/templates/fleet-server.yaml#L63)?